esque/ruleparser/fieldeval.py

```diff
@@ -28,25 +30,46 @@ def evaluate_field(self, field_name: str):
+                elif field_name == Operator.FIELDS["MESSAGE_PARTITION"].replace("\\", ""):
+                    if isinstance(self.__message, Message):
+                        partition = self.__message.partition()
+                    elif isinstance(self.__message, KafkaMessage):
+                        partition = self.__message.partition
+                    return -1 if not partition else partition
+                elif field_name == Operator.FIELDS["MESSAGE_TIMESTAMP"].replace("\\", "") and isinstance(
+                    self.__message, Message
+                ):
                     return datetime.datetime.strftime(
                         datetime.datetime.fromtimestamp(self.__message.timestamp()[1] / 1000.0), self.__time_format
                     )
                 elif field_name == Operator.FIELDS["MESSAGE_LENGTH"].replace("\\", ""):
-                    return len(self.__message)
+                    if isinstance(self.__message, Message):
+                        return len(self.__message)
+                    elif isinstance(self.__message, KafkaMessage):
+                        return len(self.__message.key) + len(self.__message.value)
                 elif field_name == Operator.FIELDS["MESSAGE_KEY"].replace("\\", ""):
-                    return self.__message.key().decode("UTF-8")
-                elif self.__header_pattern.fullmatch(field_name):
+                    if isinstance(self.__message, Message):
+                        return self.__message.key().decode("UTF-8")
+                    elif isinstance(self.__message, KafkaMessage):
+                        return self.__message.key
+                elif self.__header_pattern.fullmatch(field_name) and isinstance(self.__message, Message):
                     header_name = field_name.split(".")[-1]
-                    header_value = self.__message.headers()[header_name]
-                    if header_value is None:
+                    if isinstance(self.__message, Message):
+                        if self.__message.headers() is None:
+                            return ""
+                        decoded_message = decode_message(message=self.__message)
+                        for message_header in decoded_message.headers:
+                            if message_header.key == header_name:
+                                return message_header.value if message_header.value else ""
+                    elif isinstance(self.__message, KafkaMessage):
+                        for message_header in self.__message.headers:
+                            if message_header.key == header_name and message_header.value:
+                                return message_header.value
+                            elif message_header.key == header_name:
+                                return ""
                         return ""
-                    else:
-                        return header_value
+                return -1
```

Comment:
You can dedent this by two levels in order to satisfy flake8 :)
```python
if a:
    print('a')
else:
    if b:
        print('b')
    else:
        print('c')
```
is the same as
```python
if a:
    print('a')
elif b:
    print('b')
else:
    print('c')
```

Reply:
Simplified.